aptos-indexer-processors-sdk/sdk/src/server_framework/mod.rs

```diff
@@ -28,24 +30,57 @@ pub struct ServerArgs {
     pub config_path: PathBuf,
 }
 
+// impl ServerArgs {
+//     pub async fn run<C, P>(&self, processor: P, handle: Handle) -> Result<()>
+//     where
+//         C: RunnableConfig,
+//         P: ProcessorTrait + 'static,
+//     {
+//         // Set up the server.
+//         setup_logging();
+//         setup_panic_handler();
+//         let config = load::<ServerConfig>(&self.config_path)?;
+//         run_server_with_config(config, processor, handle).await
+//     }
+
+//     pub fn get_processor_name(&self) -> String {
+//         let config = load::<ServerConfig>(&self.config_path)?;
+//         config.server_config.processor_name
+//     }
+// }
+
 impl ServerArgs {
-    pub async fn run<C>(&self, handle: Handle) -> Result<()>
+    pub async fn run<P, D>(&self, processor: P, handle: Handle) -> Result<()>
     where
-        C: RunnableConfig,
+        P: ProcessorTrait + 'static,
+        D: DbConfig + Send + Sync + 'static,
```

Comment:
Here too. I don't think we need a DbConfig: RunnableDbConfig: ProcessorStatusSaver just to run a processor